### ISSUE TITLE
ModPlug restarts music for mod, s3m, it, and xm modules

### ISSUE THREAD

**Issue by Kyuchumimo** [pygame/pygame#3103]
**Environment:**

- **Operating system** : Linux (Lubuntu 21.10 x86_64)
- **Python version** : 3.7.9 / 3.10.3
- **SDL version** : SDL 2.0.16
- **PyGame version** : 2.0.2 / 2.1.2
- **Relevant hardware** : NA

**Current behavior:**
ModPlug restarts music for mod, s3m, it, and xm music modules
In linux it happens since Pygame 2.0.2, because in previous versions MikMod was used, while in Windows it happens in any Pygame version.

**Expected behavior:**

Music loop

**Screenshots**

Pygame 2.0.1 vs Pygame 2.0.2:

https://user-images.githubusercontent.com/74131798/160991279-ed61f104-05d3-4960-90d9-6588db35c0ef.mp4

**Steps to reproduce:**

1. Just load a music module (mod, s3m, it or xm) containing a loop and play it.

**Test code**

```py
pygame.mixer.music.load("assets/music/music.xm")) #mod, s3m, it, and xm are supported
pygame.mixer.music.play(-1)
```

**Stack trace/error output/other error logs**

NA

**Comment by ankith26** [comment]
I believe this bug could be a consequence of #2742 where I added `libmodplug` to our mac/linux buildscripts. Till that point I believe only `libmikmod` was compiled in with `SDL_mixer`. This is still a guess, but it's the only thing related to modplug/mikmod I could find that changed in 2.0.2

This could potentially be an `SDL_mixer` bug, but for now, making a temporary `libmodplug` revert could also work, IG?

**Comment by robertpfeiffer** [comment]
There have been some changes in SDL_mixer and SDL_sound recently, and they may either fix this and we need to update, or alternatively they could have caused this. Do we have a way to check which SDL_mixer version we use?

**Comment by Starbuck5** [comment]
@robertpfeiffer 

SDL_mixer last released in 2018, so the SDL_mixer version is constant between all of our modern versions (SDL_mixer 2.0.4)

Also I had been wondering about the difference between SDL_mixer and SDL_sound, I think I saw you saying something like that too. From my understanding, SDL_sound is not in active use by us in any way, even as a sub dependency. I believe SDL_mixer and SDL_sound are fully independent from each other. SDL_sound just decodes audio, SDL_mixer actually does some fancier stuff to pass it to SDL's audio playing utilities.

**Comment by Kyuchumimo** [comment]
I have tested this behavior on Windows and it occurs in all versions of Pygame, even in version 2.0.1

**Comment by Kyuchumimo** [comment]
I have compared the audio from the video with the audio you get from using MikWin and ModPlug Player and I am pretty sure that Pygame 2.0.1 uses MikMod, while Pygame 2.0.2 uses ModPlug.

![mikmod-modplug_playback](https://user-images.githubusercontent.com/74131798/179629592-ed120767-c560-491e-8d83-e3eef1a573be.png)

**Comment by Kyuchumimo** [comment]
You should consider using MikWin for Pygame on Windows and MikMod for Pygame on Linux for music module playback instead of modplug. This could potentially solve the problem.

I have attached some links to MikWin's project:
http://www.bitplanet.de/mikwin/index.html
http://ozzysrepository.blogspot.com/

**Comment by Starbuck5** [comment]
Looking at the current SDL_mixer main, they now support either libmodplug or libxmp. https://github.com/libsdl-org/SDL_mixer/blob/main/configure#L1662-L1671.

We always use SDL_mixer prebuilts on Windows. On Linux, we control it, but it looks like MikMod is not an option anymore.

It appears libxmp was made the default MOD backend of SDL_mixer 3 days ago. Best care scenario for you, libxmp solves your problem and will gradually be updated across pygame buildconfig solutions.

**Comment by Kyuchumimo** [comment]
Solved since Pygame version 2.1.3.dev6, although it requires the use of the Bxx command in the last pattern.
